### PR TITLE
types: fix incorrect time fraction parsing method

### PR DESCRIPTION
**Pull request by WangXiangUSTC**
### What problem does this PR solve? 
fix issue https://github.com/pingcap/tidb/issues/9734
tidb don't support date time format like "2019.01.01"

### What is changed and how it works?
some minor update, and add unit test

### Check List 

Tests 

 - Unit test

Related changes

 - Need to cherry-pick to the release branch

### CODE REVIEW

File: types/time.go [hunk 1]
```diff
@@ -399,7 +399,13 @@ func (t Time) RoundFrac(sc *stmtctx.StatementContext, fsp int) (Time, error) {
 
 // GetFsp gets the fsp of a string.
 func GetFsp(s string) (fsp int) {
-	fsp = len(s) - strings.LastIndex(s, ".") - 1
+	index := getFracIndex(s)
+	if index < 0 {
+		fsp = 0
+	} else {
+		fsp = len(s) - getFracIndex(s) - 1
```
Comment (zz-jason) [inline types/time.go]:
```suggestion
		fsp = len(s) - index - 1
```

File: types/time.go [hunk 2]
```diff
@@ -408,6 +414,23 @@ func GetFsp(s string) (fsp int) {
 	return
 }
 
+// find the last '.' for get fracStr, index = -1 means fracStr not found.
+// for format like '2019.01.01 00:00:00', the fracStr should be empty
+func getFracIndex(s string) (index int) {
```
Comment (winkyao) [inline types/time.go]: Please use golang comment style. comments start with `getFracIndex`

**Comment by winkyao** [comment]
Any reference mysql doc?

**Comment by WangXiangUSTC** [comment]
@winkyao I search the mysql's document, can't find the detail about the format of date time string. I just execute in mysql to check.

**Comment by zz-jason** [comment]
@WangXiangUSTC Please cherry pick this PR to release 2.1

**Pull request merged**